pkg/agent/apiserver/handlers/ovsflows/handler_test.go

```diff
@@ -171,43 +172,126 @@ func TestServiceFlows(t *testing.T) {
 
 func TestNetworkPolicyFlows(t *testing.T) {
 	ctrl := gomock.NewController(t)
-	testNetworkPolicy := &cpv1beta.NetworkPolicy{}
+	testNetworkPolicy := &cpv1beta.NetworkPolicy{
+		SourceRef: &cpv1beta.NetworkPolicyReference{
+			Type:      cpv1beta.K8sNetworkPolicy,
+			Namespace: "default",
+		},
+	}
+	testANNP := &cpv1beta.NetworkPolicy{
+		SourceRef: &cpv1beta.NetworkPolicyReference{
+			Type:      cpv1beta.AntreaNetworkPolicy,
+			Namespace: "default",
+		},
+	}
+	testACNP := &cpv1beta.NetworkPolicy{
+		SourceRef: &cpv1beta.NetworkPolicyReference{
+			Type: cpv1beta.AntreaClusterNetworkPolicy,
+		},
+	}
+	testANP := &cpv1beta.NetworkPolicy{
+		SourceRef: &cpv1beta.NetworkPolicyReference{
+			Type: cpv1beta.AdminNetworkPolicy,
+		},
+	}
 	testcases := []testCase{
 		{
 			test:           "Existing NetworkPolicy",
 			name:           "np1",
-			namespace:      "ns1",
-			query:          "?networkpolicy=np1&&namespace=ns1",
+			namespace:      "default",
+			policyType:     cpv1beta.K8sNetworkPolicy,
+			query:          "?networkpolicy=np1&namespace=default&type=K8sNP",
 			expectedStatus: http.StatusOK,
 		},
 		{
 			test:           "Non-existing NetworkPolicy",
 			name:           "np2",
 			namespace:      "ns2",
-			query:          "?networkpolicy=np2&&namespace=ns2",
+			policyType:     cpv1beta.K8sNetworkPolicy,
+			query:          "?networkpolicy=np2&namespace=ns2&type=K8sNP",
 			expectedStatus: http.StatusNotFound,
 		},
+		{
+			test:           "Ambiguous query",
+			name:           "np3",
+			namespace:      "ns3",
+			query:          "?networkpolicy=np3&namespace=ns3",
+			expectedStatus: http.StatusBadRequest,
+			expectedErr:    errAmbiguousQuery,
+		},
+		{
+			test:           "Existing ACNP",
+			name:           "acnp1",
+			policyType:     cpv1beta.AntreaClusterNetworkPolicy,
+			query:          "?networkpolicy=acnp1&type=ACNP",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			test:           "ACNP bad request",
+			name:           "acnp2",
+			policyType:     cpv1beta.AntreaClusterNetworkPolicy,
+			query:          "?networkpolicy=acnp2&type=ACNP&namespace=default",
+			expectedStatus: http.StatusBadRequest,
+		},
+		{
+			test:           "Existing ANNP",
+			name:           "annp1",
+			namespace:      "default",
+			policyType:     cpv1beta.AntreaNetworkPolicy,
+			query:          "?networkpolicy=annp1&namespace=default&type=ANNP",
+			expectedStatus: http.StatusOK,
+		},
+		{
+			test:           "ANNP bad request",
+			name:           "annp2",
+			policyType:     cpv1beta.AntreaNetworkPolicy,
+			query:          "?networkpolicy=annp2&type=ANNP",
+			expectedStatus: http.StatusBadRequest,
+		},
+		{
+			test:           "Existing ANP",
+			name:           "anp1",
+			policyType:     cpv1beta.AdminNetworkPolicy,
+			query:          "?networkpolicy=anp1&type=ANP",
+			expectedStatus: http.StatusOK,
+		},
 	}
 	for i := range testcases {
```

Comment:
not introduced by you, but each testcase should be its own subtest defined with `t.Run`, and each subtest should have its own mock controller